components/tools/OmeroM/src/helper/strToMapAnnotation.m

```diff
@@ -66,29 +76,18 @@
 end
 
 
-import java.util.ArrayList
-eval('import omero.model.NamedValue')
-
-li = ArrayList;
-
-for r = 1:size(str,1)
-    
-    li.add(NamedValue(str{r,1},str{r,2}));
-
-end
-
-eval('import omero.model.MapAnnotationI')
-
-ma = MapAnnotationI(int64(1),true); % 'false' results in Java exception occurred: omero.UnloadedEntityException:
-ma.setMapValue(li);
-
 if iseditable
     %NOTE this is required to make it editable from GUI
     eval('import omero.constants.metadata.NSCLIENTMAPANNOTATION')
-    ma.setNs(rstring(NSCLIENTMAPANNOTATION.value));
+    namespace = char(NSCLIENTMAPANNOTATION.value);
+else
+    namespace = '';
 end
 
 
+ma = writeMapAnnotation(session,...
```

Comment:
`strToMapAnnotation` is now a wrapper of `writeMapAnnotation`. This solves the problem about MapAnnotation ID assignment mentioned above.